src/runtime_src/core/edge/user/aie/aie.h

```diff
@@ -77,8 +87,16 @@ class Aie {
     XAieGbl_Config *aieConfigPtr; // AIE configuration pointer
     XAieGbl aieInst;              // AIE global instance
     XAieGbl_HwCfg aieConfig;      // AIE configuration pointer
+
+    void
+    submit_sync_bo(uint64_t paddr, std::vector<gmio_type>::iterator gmio, enum xclBOSyncDirection dir, size_t size);
+
+    /* Wait for all the BD trasters for a given channel */
+    void
+    wait_sync_bo(ShimDMA *dmap, uint32_t chan, uint32_t timeout);
```

Comment:
Does this function change dmap?   If not it should be const ShimDMA*

Reply:
The content of dmap pointing to will change. I am using 
wait_sync_bo(ShimDMA* const dmap, uint32_t chan, uint32_t timeout);